### PR TITLE
o2-eve, o2-eve-export-workflow, o2-eve-convert: serialisation to root…

### PR DESCRIPTION
**Pull request by jmyrcha**
… files, conversion tool

### CODE REVIEW

**Comment by jmyrcha** [comment]
there is no trailing spaces in O2DPLDisplay.cxx

File: EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h [hunk 1]
```diff
@@ -33,10 +34,11 @@ class FileWatcher
   std::string prevItem(const std::string& item) const;
   std::string mDataFolder;  ///< folder being observed
   std::string mCurrentFile; ///< "current" file name
+  std::vector<std::string> mExt; ///< extensions of files to be observed
   bool currentFileExist();
 
  public:
-  FileWatcher(const std::string& path);
+  FileWatcher(const std::string& path, std::vector<std::string> ext);
```
Comment (shahor02) [inline EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h:41]: why passing the vector by copy?

**Comment by jmyrcha** [comment]
fullCI failed on irrelevant problems

File: EventVisualisation/DataConverter/src/VisualisationEventROOTSerializer.cxx [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+  calo.Branch("GID", &calo_GID);
+  calo.Branch("PID", &calo_PID);
+
+  for (auto calorimeter : event.getCalorimetersSpan()) {
+    calo_source = calorimeter.getSource();
+    calo_time = calorimeter.getTime();
+    calo_energy = calorimeter.getEnergy();
+    calo_eta = calorimeter.getEta();
+    calo_phi = calorimeter.getPhi();
+    calo_GID = calorimeter.getGIDAsString();
```
Comment (shahor02) [inline EventVisualisation/DataConverter/src/VisualisationEventROOTSerializer.cxx:178]: what is the reason of using string for GID, which is a simple integer?

**Review by shahor02 — COMMENTED** [review]
Hi,
Please see comments below. Also in plenty of places you are passing string by copy, please use `const std::string&` instead.
If you prefer, I can merge it as is now and you address questions later.

File: EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h (hunk 1 above)
Comment (jmyrcha) [reply EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h:41]: It is executed once in a program lifetime (may be executed again on source switch, but seldom done) so I prefer code readibility over efficiency - passing by reference will require creating variable just over place where it is called. If you request I can refactor it in next PR (so please merge it as it is now with request to refactor)

File: EventVisualisation/DataConverter/src/VisualisationEventROOTSerializer.cxx (hunk 2 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/src/VisualisationEventROOTSerializer.cxx:178]: It was preserved from the previous version to keep interoperability with previous *.json files. I will address that in next (separate to that) PR when I will be able to test this change only. Please merge as it is and I will fix it soon.

File: EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h (hunk 1 above)
Comment (shahor02) [reply EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h:41]: Not sure what you mean, now you are creating a copy to pass a vector, by passing the reference you create nothing.

Comment (jmyrcha) [reply EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h:41]: And such variable will be visible in outer space - when it has no usage.
something like:
static std::vector<std::string> extensions = ...

Comment (shahor02) [reply EventVisualisation/Base/include/EventVisualisationBase/FileWatcher.h:41]: there is some misunderstanding: as it is now you are creating 2 clones of the vector: 1st one just to pass the vector, 2nd as a data member of the class.
To get rid of the 1st copy you just need to define your constructor as:
```
 FileWatcher(const std::string& path, const std::vector<std::string>& ext) : mExt(ext) ... {}
```
To get rid of the 2nd one: define your class as 
```
class FileWatcher
{
....
 const std::vector<std::string>& mExt;
};
```

**Review by shahor02 — APPROVED** [review]
Merging but please get rid of excessive copies I mentioned.

**Review by shahor02 — CHANGES_REQUESTED** [review]
there are actually conflicts, please rebase to dev and resubmit.

**Review by shahor02 — APPROVED** [review]
Tests were passed, merging after rebase.

**Pull request merged**